modules/expo-bluesky-oauth-client/src/oauth-client-react-native.ts

```diff
@@ -0,0 +1,48 @@
+import {requireNativeModule} from 'expo-modules-core'
+import {Jwk, Jwt, Key} from '@atproto/jwk'
+
+const NativeModule = requireNativeModule('ExpoBlueskyOAuthClient')
+
+const LINKING_ERROR =
+  'The package ExpoBlueskyOAuthClient is not linked. Make sure you have run `expo install expo-bluesky-oauth-client` and rebuilt your app.'
+
+export const OauthClientReactNative = (NativeModule as null) || {
+  getRandomValues(_length: number): Uint8Array {
+    throw new Error(LINKING_ERROR)
+  },
+
+  /**
+   * @throws if the algorithm is not supported ("sha256" must be supported)
+   */
+  async digest(_bytes: Uint8Array, _algorithm: string): Promise<Uint8Array> {
+    throw new Error(LINKING_ERROR)
+  },
+
+  /**
+   * Create a private JWK for the given algorithm. The JWK should have a "use"
+   * an does not need a "kid" property.
+   *
+   * @throws if the algorithm is not supported ("ES256" must be supported)
+   */
+  async generateJwk(_algo: string): Promise<{publicKey: Key; privateKey: Key}> {
```

Comment:
@haileyok You don't need to generate a key pair in the native code as the `@atproto/jwk` lib already handles the conversion from private to public (and only requires a private key as input).

Also, the native code will generate a `Jwk`, not a `Key` instance. The `Key` instance needs to be instantiated using the jwk:

```suggestion 
  async generateJwk(_algo: string): Promise<Jwk> {
```

This can then be used as:

```ts
const jwk = await OauthClientReactNative.generateJwk(algo)
const kid = "random string"
return new ReactNativeKey({ ...jwk, kid })
```

Reply:
Thanks for that tip! And yea, I'm just pushing up some code right now with the skeleton you have over in the oauth feature branch 👍